demo_modules/sliced_cubes/sliced_cubes_client.ts

```diff
@@ -0,0 +1,539 @@
+import { Client } from '../../client/modules/module_interface.ts';
+import {
+  CurrentValueInterpolator,
+  ModuleState,
+  NumberLerpInterpolator,
+  SharedState,
+} from '../../client/network/state_manager.ts';
+import { Three, ThreeJsSurface } from '../../client/surface/threejs_surface.ts';
+import { Polygon } from '../../lib/math/polygon2d.ts';
+import { PieceSetState, SlicedCubesState, STATE_NAME } from './sliced_cubes.ts';
+
+/**
+ * Loads the SlicedCubes client.
+ */
+export function load(state: ModuleState, wallGeometry: Polygon) {
+  class SlicedCubesClient extends Client {
+    private sharedState!: SharedState;
+    private slices!: CubeSlices;
+
+    override willBeShownSoon(container: HTMLElement): Promise<void> {
+      const surface = new ThreeJsSurface(container, wallGeometry, {
+        antialias: true,
+      });
+      this.surface = surface;
+      this.slices = new CubeSlices(surface);
+      this.sharedState = state.define(STATE_NAME, [
+        {
+          co: NumberLerpInterpolator,
+          cn: NumberLerpInterpolator,
```

Comment:
Should this be current value interpolator?

Reply:
No. Despite the similarity of the names, `cn` is a real number and `an` and `bn` are integers.